src/variables.jl

```diff
@@ -124,6 +124,65 @@ function add_constrained_variable(model::ModelLike, set::AbstractScalarSet)
     return variable, constraint
 end
 
+"""
+    add_constrained_variable(
+        model::ModelLike,
+        set::Tuple{<:GreaterThan,<:LessThan},
+    )
+
+A special-case method to add a scalar variable with a lower and upper bound.
+
+This method should be implemented by optimizers which have native support for
+adding a variable with bounds and which cannot performantly modify the variable
+bounds after creation.
+
+## Example
+
+```jldoctest
+julia> import MathOptInterface as MOI
+
+julia> model = MOI.Utilities.Model{Float64}();
+
+julia> set = (MOI.GreaterThan(1.0), MOI.LessThan(2.0));
+
+julia> x, (c_l, c_u) = MOI.add_constrained_variable(model, set);
+
+julia> c_l
+MathOptInterface.ConstraintIndex{MathOptInterface.VariableIndex, MathOptInterface.GreaterThan{Float64}}(1)
+
+julia> c_u
+MathOptInterface.ConstraintIndex{MathOptInterface.VariableIndex, MathOptInterface.LessThan{Float64}}(1)
+
+julia> print(model)
+Feasibility
+
+Subject to:
+
+VariableIndex-in-GreaterThan{Float64}
+ v[1] >= 1.0
+
+VariableIndex-in-LessThan{Float64}
+ v[1] <= 2.0
+```
+"""
+function add_constrained_variable(
+    model::ModelLike,
+    set::Tuple{<:GreaterThan,<:LessThan},
```

Comment:
@blegat are you okay with this signature?

Reply:
I have a small preference for two separate arguments rather than the tuple. But if you both prefer the Tuple, let's do it!

Reply:
I feel like the two separate arguments just opens the door too much.

This feels more like a "set" composed of a tuple. Rather than a completely new method. It's more logically consistent with the rest of the API. And the output of the constraint index matches the input of the set. Tuple->Tuple.

Reply:
I also think restricting to this one special case is a good start. We can always relax later if it turns out to be a good idea. But this is really the only case where we've had the need, and it's been years since we started MOI...

Reply:
Fair enough, lets go for it.

Reply:
I continued to think about this all day, and I didn't come up with a reason not to do this.

Reply:
> And the output of the constraint index matches the input of the set.

I agree, otherwise `MOI.add_constrained_variable(model, set)` should return `vi, (ci,)` but that would be breaking in addition to being annoying